### ISSUE TITLE
Attaching pdf file fails with duplicate key value violates unique constraint "memgpt_archival_memory_agent_pkey"

### ISSUE THREAD

**Issue by haikyuu** [cpacker/MemGPT#949]
**Describe the bug**
Attaching a pdf file to my agent breaks

**Please describe your setup**

- [x] MemGPT version: 0.3.0
- [x] How did you install memgpt? From source + using poetry with postgres extra
- [x] Describe your setup
  - What's your OS (Windows/MacOS/Linux)? MacOS
  - How are you running `memgpt`? in poetry shell

```
m run --agent s3

🔁 Using existing agent s3

Hit enter to begin (will request first MemGPT message)

💭 The initial greeting went well. Moving forward, I should strive to help Abdellah with his daily tasks, answer his queries and ultimately, assist him to become better at what he does. Let's see what he needs assistance with first.
🤖 What can I assist you with today, Abdellah?
> Enter your message: /attach
Warning: the following sources are not compatible with this agent's embedding model and dimension: []
? Select data source krl_research_paper
Ingesting 257 passages into s3
  0%|                                                                                                                                                                                                                   | 0/3 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/pg8000/legacy.py", line 254, in execute
    self._context = self._c.execute_unnamed(
                    ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/pg8000/core.py", line 707, in execute_unnamed
    self.handle_messages(context)
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/pg8000/core.py", line 827, in handle_messages
    raise context.error
pg8000.exceptions.DatabaseError: {'S': 'ERROR', 'V': 'ERROR', 'C': '23505', 'M': 'duplicate key value violates unique constraint "memgpt_archival_memory_agent_pkey"', 'D': 'Key (id)=(25d04eea-af7f-43ce-bcb6-870fb0d3ff2e) already exists.', 's': 'public', 't': 'memgpt_archival_memory_agent', 'n': 'memgpt_archival_memory_agent_pkey', 'F': 'nbtinsert.c', 'L': '673', 'R': '_bt_check_unique'}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1938, in _exec_single_context
    self.dialect.do_executemany(
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/sqlalchemy/engine/default.py", line 919, in do_executemany
    cursor.executemany(statement, parameters)
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/pg8000/legacy.py", line 306, in executemany
    self.execute(operation, parameters)
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/pg8000/legacy.py", line 281, in execute
    raise cls(msg)
pg8000.dbapi.IntegrityError: {'S': 'ERROR', 'V': 'ERROR', 'C': '23505', 'M': 'duplicate key value violates unique constraint "memgpt_archival_memory_agent_pkey"', 'D': 'Key (id)=(25d04eea-af7f-43ce-bcb6-870fb0d3ff2e) already exists.', 's': 'public', 't': 'memgpt_archival_memory_agent', 'n': 'memgpt_archival_memory_agent_pkey', 'F': 'nbtinsert.c', 'L': '673', 'R': '_bt_check_unique'}

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/bin/memgpt", line 6, in <module>
    sys.exit(app())
             ^^^^^
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/typer/main.py", line 328, in __call__
    raise e
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/typer/main.py", line 311, in __call__
    return get_command(self)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/typer/core.py", line 778, in main
    return _main(
           ^^^^^^
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/typer/core.py", line 216, in _main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/typer/main.py", line 683, in wrapper
    return callback(**use_params)  # type: ignore
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/abdellah/workspace/scrimba/MemGPT/memgpt/cli/cli.py", line 653, in run
    run_agent_loop(memgpt_agent, config, first, ms, no_verify)  # TODO: add back no_verify
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/abdellah/workspace/scrimba/MemGPT/memgpt/main.py", line 146, in run_agent_loop
    attach(memgpt_agent.agent_state.name, data_source)
  File "/Users/abdellah/workspace/scrimba/MemGPT/memgpt/cli/cli.py", line 733, in attach
    dest_storage.insert_many(passages)
  File "/Users/abdellah/workspace/scrimba/MemGPT/memgpt/agent_store/db.py", line 365, in insert_many
    session.commit()
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 1969, in commit
    trans.commit(_to_root=True)
  File "<string>", line 2, in commit
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/sqlalchemy/orm/state_changes.py", line 139, in _go
    ret_value = fn(self, *arg, **kw)
                ^^^^^^^^^^^^^^^^^^^^
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 1256, in commit
    self._prepare_impl()
  File "<string>", line 2, in _prepare_impl
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/sqlalchemy/orm/state_changes.py", line 139, in _go
    ret_value = fn(self, *arg, **kw)
                ^^^^^^^^^^^^^^^^^^^^
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 1231, in _prepare_impl
    self.session.flush()
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 4312, in flush
    self._flush(objects)
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 4447, in _flush
    with util.safe_reraise():
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/sqlalchemy/util/langhelpers.py", line 146, in __exit__
    raise exc_value.with_traceback(exc_tb)
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 4408, in _flush
    flush_context.execute()
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/sqlalchemy/orm/unitofwork.py", line 466, in execute
    rec.execute(self)
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/sqlalchemy/orm/unitofwork.py", line 642, in execute
    util.preloaded.orm_persistence.save_obj(
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/sqlalchemy/orm/persistence.py", line 93, in save_obj
    _emit_insert_statements(
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/sqlalchemy/orm/persistence.py", line 1044, in _emit_insert_statements
    result = connection.execute(
             ^^^^^^^^^^^^^^^^^^^
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1416, in execute
    return meth(
           ^^^^^
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/sqlalchemy/sql/elements.py", line 517, in _execute_on_connection
    return connection._execute_clauseelement(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1639, in _execute_clauseelement
    ret = self._execute_context(
          ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1848, in _execute_context
    return self._exec_single_context(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1988, in _exec_single_context
    self._handle_dbapi_exception(
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 2344, in _handle_dbapi_exception
    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1938, in _exec_single_context
    self.dialect.do_executemany(
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/sqlalchemy/engine/default.py", line 919, in do_executemany
    cursor.executemany(statement, parameters)
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/pg8000/legacy.py", line 306, in executemany
    self.execute(operation, parameters)
  File "/Users/abdellah/Library/Caches/pypoetry/virtualenvs/pymemgpt-kDMo5dIc-py3.11/lib/python3.11/site-packages/pg8000/legacy.py", line 281, in execute
    raise cls(msg)
sqlalchemy.exc.IntegrityError: (pg8000.dbapi.IntegrityError) {'S': 'ERROR', 'V': 'ERROR', 'C': '23505', 'M': 'duplicate key value violates unique constraint "memgpt_archival_memory_agent_pkey"', 'D': 'Key (id)=(25d04eea-af7f-43ce-bcb6-870fb0d3ff2e) already exists.', 's': 'public', 't': 'memgpt_archival_memory_agent', 'n': 'memgpt_archival_memory_agent_pkey', 'F': 'nbtinsert.c', 'L': '673', 'R': '_bt_check_unique'}
[SQL: INSERT INTO memgpt_archival_memory_agent (id, user_id, text, doc_id, agent_id, data_source, embedding, embedding_dim, embedding_model) VALUES (%s::UUID, %s::UUID, %s::VARCHAR, %s::UUID, %s::UUID, %s::VARCHAR, %s, %s::BIGINT, %s::VARCHAR)]
[parameters: [(UUID('25d04eea-af7f-43ce-bcb6-870fb0d3ff2e'), UUID('00000000-0000-0000-0000-d2c1cd7f45ce'), 'An Overview of KRL, a Knowledge \nRepresentation Language* \nDANIEL G. BOBROW \nXerox Palo Alto Research Center \nTERRY WINOGRAD \nStanford Universit ... (1007 characters truncated) ... ject. This procedural attachment allows the steps for a particular operation to be \ndetermined by characteristics of the specific entities involved.', None, UUID('c7c0239a-c758-4fb2-92d9-54245b499f0d'), 'krl_research_paper', '[-0.003979679197072983,0.03137205168604851,0.009362339973449707,-0.029072530567646027,-0.005204722285270691,0.012284649536013603,-0.00239191367290914 ... (42801 characters truncated) ... ,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0]', 1536, 'text-embedding-ada-002'), (UUID('45bed0db-7e0a-4231-a746-5de4dd63be9c'), UUID('00000000-0000-0000-0000-d2c1cd7f45ce'), 'It outlines both the general concepts which underlie our research and \nthe details of KRL-0, an experimental implementation of some of these concept ... (1079 characters truncated) ... ess- \ning. It provides for a priority-ordered multiprocess agenda with explicit (user-provided) \nstrategies for scheduling and resource allocation.', None, UUID('c7c0239a-c758-4fb2-92d9-54245b499f0d'), 'krl_research_paper', '[-0.002274206141009927,0.025078481063246727,0.0109217194840312,-0.03273750841617584,-0.014543860219419003,0.015829581767320633,0.004496568813920021,- ... (42788 characters truncated) ... ,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0]', 1536, 'text-embedding-ada-002'), (UUID('6fde6684-eaf6-4b4e-99f5-536c07f7825e'), UUID('00000000-0000-0000-0000-d2c1cd7f45ce'), 'The formalism for declarative knowledge is based on structured conceptual \nobjects with associated descriptions. These objects form a network of mem ... (1201 characters truncated) ... standing, and through these to develop theories of human \nlanguage use. What we describe is a formal computer language for representing \nknowledge.', None, UUID('c7c0239a-c758-4fb2-92d9-54245b499f0d'), 'krl_research_paper', '[-0.002843925729393959,0.027784541249275208,0.006594907026737928,-0.029325854033231735,-0.010093549266457558,0.014540167525410652,-0.0014364552916958 ... (42795 characters truncated) ... ,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0]', 1536, 'text-embedding-ada-002'), (UUID('142608fc-33d8-4f34-af55-a88d2db155e4'), UUID('00000000-0000-0000-0000-d2c1cd7f45ce'), 'The control structure of KRL is based on the belief that the next generation of intelligent \nprograms will integrate data-directed and goal-directed ... (1022 characters truncated) ... of memory and attention. \nOur ideas are in the course of active expansion and modification, and there \n*Requests for reprints should be sent to Dr.', None, UUID('c7c0239a-c758-4fb2-92d9-54245b499f0d'), 'krl_research_paper', '[-0.011057169176638126,0.016662249341607094,-0.007816514000296593,-0.01748284511268139,-4.645184526452795e-05,0.011975122615695,-0.001360414316877722 ... (42825 characters truncated) ... ,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0]', 1536, 'text-embedding-ada-002'), (UUID('5d5e7e32-7b0c-4a5a-b246-f0beaa5621f8'), UUID('00000000-0000-0000-0000-d2c1cd7f45ce'), 'It provides procedure directories which \noperate along with process frameworks to allow procedural parameterization of the \nfundamental system proc ... (822 characters truncated) ... hould be sent to Dr. Daniel G. Bobrow, Computer Science Laboratory, Xerox Palo Alto Research Center, 3333 Coyote Hill Road, Palo Alto, CA. 94304. \n3', None, UUID('c7c0239a-c758-4fb2-92d9-54245b499f0d'), 'krl_research_paper', '[-0.003062684088945389,0.02786523476243019,0.0003756979131139815,-0.01925511285662651,0.004246229771524668,0.01406412199139595,-0.003671760205179453, ... (42799 characters truncated) ... ,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0]', 1536, 'text-embedding-ada-002'), (UUID('cc80ac36-3fc9-43ce-b1b1-b00f197642d7'), UUID('00000000-0000-0000-0000-d2c1cd7f45ce'), "AN OVERVIEW OF KRL 17 \n[Person UNIT Basic \n<SELF> ... There is no description for Person since it is not further \nanalyzed in the travel planning  ... (258 characters truncated) ... \ndescriptors. Unless otherwise specified, the hometown of a \nPerson will be assumed to be PaloAlto (which is in turn a unit \ndescribed as a City).", None, UUID('c7c0239a-c758-4fb2-92d9-54245b499f0d'), 'krl_research_paper', '[0.012006288394331932,0.025723785161972046,0.011408060789108276,-0.020951878279447556,-0.010441157966852188,0.025682048872113228,-0.00399977760389447 ... (42810 characters truncated) ... ,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0]', 1536, 'text-embedding-ada-002'), (UUID('34055945-1741-46c5-ba81-81a0def5ba02'), UUID('00000000-0000-0000-0000-d2c1cd7f45ce'), '4 D. G. BOBROW AND T. WINOGRAD \nwill be many changes before KRL comes close to our goals. We have \nimplemented a subsetof our ideas in a system whi ... (719 characters truncated) ... e a \nhouse of cards. The researchers are interested in the higher levels, and try to \nbuild up the minimum of supporting props at the lower levels.', None, UUID('c7c0239a-c758-4fb2-92d9-54245b499f0d'), 'krl_research_paper', '[-0.000691378372721374,0.012739204801619053,-0.0018100732704624534,-0.0017422735691070557,0.0006864718161523342,0.017642192542552948,-0.0055631427094 ... (42834 characters truncated) ... ,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0]', 1536, 'text-embedding-ada-002'), (UUID('7e637a89-004b-4d25-b706-9a44e5dccc97'), UUID('00000000-0000-0000-0000-d2c1cd7f45ce'), 'We have con- \nducted some experiments in system building in KRL-0 to test its utility and \nhabitability. Although we have planned a number of featu ... (893 characters truncated) ... e heights, but collapses immediately if swayed in the slightest from the \nspecific domain (often even the specific examples) for which it was built.', None, UUID('c7c0239a-c758-4fb2-92d9-54245b499f0d'), 'krl_research_paper', '[0.006126410327851772,0.026844946667551994,0.0031113463919609785,-0.01448514498770237,0.0018044026801362634,0.024634018540382385,-0.00062583532417193 ... (42809 characters truncated) ... ,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0]', 1536, 'text-embedding-ada-002')  ... displaying 10 of 100 total bound parameter sets ...  (UUID('5b02ccfd-87a4-4374-a213-12f6d2e8f553'), UUID('00000000-0000-0000-0000-d2c1cd7f45ce'), 'Terminals. In applying a match process recursively to a complex structure, \nthere is a choice of where the recursion *‘bottoms out.”” In matchers op ... (973 characters truncated) ... ables in the pattern and constants (or \nother variables) in the datum, either as an explicit output or implicitly through \nside effects. \nProcess.', None, UUID('c7c0239a-c758-4fb2-92d9-54245b499f0d'), 'krl_research_paper', '[-0.01688665896654129,0.012354468926787376,0.003123736009001732,0.002951838308945298,-0.016354329884052277,0.02654251456260681,-0.01409932505339384,- ... (42814 characters truncated) ... ,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0]', 1536, 'text-embedding-ada-002'), (UUID('1993458a-5969-41b4-bab0-4a5d83e3f488'), UUID('00000000-0000-0000-0000-d2c1cd7f45ce'), 'In a Lisp-based system, \npattern variables and atoms form the terminals—the task of matching them is not \ndone by recursively setting up match subt ... (891 characters truncated) ...  sequence, and when the match as a whole should be stopped. In the \nsimplest case, the subtasks are taken serially, usually in the sequence provided', None, UUID('c7c0239a-c758-4fb2-92d9-54245b499f0d'), 'krl_research_paper', '[-0.017323087900877,0.01447674073278904,-0.001895163208246231,-0.003545324318110943,-0.010167586617171764,0.034444410353899,-0.0006147570675238967,0. ... (42821 characters truncated) ... ,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0]', 1536, 'text-embedding-ada-002')]]
(Background on this error at: https://sqlalche.me/e/20/gkpj)
```

**Comment by sarahwooders** [comment]
I believe this is a bug due to importing the same data source multiple times, and we don't have a way to check for existence/update existing important passages. This is high priority bug we should fix before the next release.

**Comment by sarahwooders** [comment]
This should be fixed by #973, but please let me know if you still have the same issue! Will close for now but can re-open if so.
